distributed/shuffle/_worker_extension.py

```diff
@@ -204,9 +229,9 @@ def _repartition_buffers(self, data: list[bytes]) -> dict[str, list[pa.Table]]:
         groups = split_by_partition(table, self.column)
         assert len(table) == sum(map(len, groups.values()))
         del data
-        return {k: [v] for k, v in groups.items()}
+        return {k: [serialize_table(v)] for k, v in groups.items()}
```

Comment:
I guess the type signature of this method is no longer accurate, is it? 

`serialize_table` is not annotated but it does return something like `tuple[dict, list[bytes]]`

Reply:
Good catch (bad mypy!)